src/software/ai/hl/stp/tactic/penalty_kick/penalty_kick_tactic_fsm.h

```diff
@@ -0,0 +1,305 @@
+
+        // Based on constant acceleration -> // dX = init_vel*t + 0.5*a*t^2
+        const double max_enemy_movement_x =
+            enemy_goalie.value().velocity().x() * time_to_pass_keeper +
+            0.5 * std::copysign(1, goalie_to_block_position.x()) *
+                PENALTY_KICK_GOALIE_MAX_ACC * pow(time_to_pass_keeper, 2);
+        const double max_enemy_movement_y =
+            enemy_goalie.value().velocity().y() * time_to_pass_keeper +
+            0.5 * std::copysign(1, goalie_to_block_position.y()) *
+                PENALTY_KICK_GOALIE_MAX_ACC * pow(time_to_pass_keeper, 2);
+
+        // If the maximum distance that the goalie can move is less than actual
+        // distance it must move to reach the ball, return true for a viable
+        // shot
+        // Not simplifying this if statement makes the code logic slightly
+        // easier to understand
+        if ((fabs(goalie_to_block_position.x()) >
+             (fabs(max_enemy_movement_x) + ROBOT_MAX_RADIUS_METERS)) ||
+            (fabs(goalie_to_block_position.y()) >
+             (fabs(max_enemy_movement_y) + ROBOT_MAX_RADIUS_METERS)))
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    /**
+     * Helper function that returns the point on the enemy goal line where the shooter
+     * should aim at.
+     *
+     * @param enemy_goalie  the goalie that the shooter is scoring against
+     * @param field         the field being played on
+     *
+     * @return the Point on the enemy goal-line where the shooter robot should aim
+     */
+    static const Point evaluateNextShotPosition(std::optional<Robot> enemy_goalie,
+                                                Field field)
+    {
+        // Evaluate if the goalie is closer to the negative or positive goalpost
+        if (enemy_goalie.has_value())
+        {
+            double goalie_dist_to_neg_goalpost =
+                (field.enemyGoalpostNeg() - enemy_goalie.value().position())
+                    .lengthSquared();
+            double goalie_dist_to_pos_goalpost =
+                (field.enemyGoalpostPos() - enemy_goalie.value().position())
+                    .lengthSquared();
+
+            return goalie_dist_to_neg_goalpost > goalie_dist_to_pos_goalpost
+                       ? field.enemyGoalpostNeg() + Vector(0, PENALTY_KICK_POST_OFFSET)
+                       : field.enemyGoalpostPos() + Vector(0, -PENALTY_KICK_POST_OFFSET);
+        }
+        else
+        {
+            // Return the center of the enemy goal
+            return Point(field.enemyGoalpostPos().x(), 0);
+        }
+    }
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        const auto approach_keeper_s = state<DribbleFSM>;
+        // const auto adjust_orientation_for_shot = state<DribbleFSM>;
+        // const auto initial_s         = state<InitialState>;
```

Comment:
Are those variables needed?

Reply:
no, this was code that I was playing around with while making PR changes, got rid of it now